assets/src/edit-story/components/panels/index.js

```diff
@@ -72,30 +75,50 @@ export function getPanels( elements ) {
 		return [];
 	}
 
+	const isBackground = ( elements.length === 1 && elements[ 0 ].isBackground );
+
 	// Panels to always display, independent of the selected element.
 	const sharedPanels = [
 		{ type: ACTIONS, Panel: ActionsPanel },
 	];
-	// Find which panels all the selected elements have in common
-	const selectionPanels = elements
-		.map( ( { type } ) => elementTypes.find( ( elType ) => elType.type === type ).panels )
-		.reduce( ( commonPanels, panels ) => intersect( commonPanels, panels ), ALL )
-		.map( ( type ) => {
-			switch ( type ) {
-				case POSITION: return { type, Panel: PositionPanel };
-				case SCALE: return { type, Panel: ScalePanel };
-				case ROTATION_ANGLE: return { type, Panel: RotationPanel };
-				case SIZE: return { type, Panel: SizePanel };
-				case FILL: return { type, Panel: FillPanel };
-				case BACKGROUND_COLOR: return { type, Panel: BackgroundColorPanel };
-				case COLOR: return { type, Panel: ColorPanel };
-				case FONT: return { type, Panel: FontPanel };
-				case STYLE: return { type, Panel: StylePanel };
-				case TEXT: return { type, Panel: TextPanel };
-				case VIDEO_POSTER: return { type, Panel: VideoPosterPanel };
-				default: throw new Error( `Unknown panel: ${ type }` );
-			}
-		} );
+
+	let selectionPanels = [];
+	// Only display background panel in case of background element.
+	if ( isBackground ) {
+		selectionPanels = [ { type: BACKGROUND, Panel: BackgroundPanel } ];
+		if ( 'video' === elements[ 0 ].type ) {
```

Comment:
I don't understand what this code is doing. Can you add a comment.

Reply:
Sure, done in 56f07cf.